runtime/dancebox/src/migrations.rs

```diff
@@ -375,6 +375,125 @@ where
     }
 }
 
+/// A reason for placing a hold on funds.
+#[derive(
+    Clone,
+    parity_scale_codec::Encode,
+    parity_scale_codec::Decode,
+    PartialEq,
+    sp_core::RuntimeDebug,
+    scale_info::TypeInfo,
+)]
+pub enum OldHoldReason {
+    /// The Pooled Stake holds
+    PooledStake,
+}
+
+pub struct MigrateHoldReasonRuntimeEnum<T>(pub PhantomData<T>);
+impl<T> Migration for MigrateHoldReasonRuntimeEnum<T>
+where
+    T: pallet_balances::Config,
+    T: pallet_pooled_staking::Config,
+    <T as pallet_balances::Config>::RuntimeHoldReason: From<pallet_pooled_staking::HoldReason>,
+{
+    fn friendly_name(&self) -> &str {
+        "TM_MigrateHoldReasonRuntimeHold"
+    }
+
+    fn migrate(&self, _available_weight: Weight) -> Weight {
+        log::info!(target: LOG_TARGET, "migrate");
+        let pallet_prefix: &[u8] = b"Balances";
+        let storage_item_prefix: &[u8] = b"Holds";
+
+        let stored_data: Vec<_> = storage_key_iter::<
+            T::AccountId,
+            BoundedVec<
+                IdAmount<OldHoldReason, <T as pallet_balances::Config>::Balance>,
+                T::MaxHolds,
+            >,
+            Blake2_128Concat,
+        >(pallet_prefix, storage_item_prefix)
+        .collect();
+
+        let migrated_count_read = stored_data.len() as u64;
+        let mut migrated_count_write = 0u64;
+
+        // Write to the new storage
+        for (account_id, holds) in stored_data {
+            let mut new_holds = vec![];
+
+            for hold in holds {
+                let new_item: pallet_balances::IdAmount<
+                    <T as pallet_balances::Config>::RuntimeHoldReason,
+                    <T as pallet_balances::Config>::Balance,
+                > = pallet_balances::IdAmount {
+                    id: pallet_pooled_staking::HoldReason::PooledStake.into(),
+                    amount: hold.amount,
+                };
+                new_holds.push(new_item);
+            }
+            let bounded = BoundedVec::<_, T::MaxHolds>::truncate_from(new_holds.clone());
+            pallet_balances::Holds::<T>::insert(&account_id, bounded);
```

Comment:
Should the old data be cleared from storage?

Reply:
we are writing into the same storage, but we are just overriding it with the new type